mountpoint-s3/src/data_cache/express_data_cache.rs

```diff
@@ -99,6 +99,30 @@ where
 
         Ok(())
     }
+
+    fn handle_get_object_err(
+        &self,
+        err: ObjectClientError<GetObjectError, Client::ClientError>,
+    ) -> DataCacheResult<Option<ChecksummedBytes>> {
+        // Error cases only. This can never return actual bytes.
+        match err {
+            ObjectClientError::ServiceError(GetObjectError::NoSuchKey) => {
+                metrics::counter!("express_data_cache.block_hit").increment(0);
+                Ok(None)
+            }
+            e => {
+                self.emit_failure_metric("get_object_failure", MetricsType::Read);
+                Err(DataCacheError::IoFailure(e.into()))
+            }
+        }
+    }
+
+    fn emit_failure_metric(&self, reason: &'static str, type_: MetricsType) {
```

Comment:
I'd suggest inlining the metrics calls. This function makes it less clear what you are tracking.